### ISSUE TITLE
Additional Information to prepare_model_for_kbit_training

### ISSUE THREAD

**Issue by NilBiescas** [huggingface/peft#2299]
### Feature request

Add a comment in the docstring of prepare_model_for_kbit_training to inform that it sets requires_grad to false to all the base model parameters.

### Motivation

As this function is used before training it might be nice to know that its actually freezing all the base model.

### Your contribution

I could add a line commenting that the function freezes the base model.

**Comment by githubnemo** [comment]
Hey @NilBiescas! I think this is a good idea since it might be surprising to find that the base model cannot be trained after using it with `prepare_model_for_kbit_training`. 

In addition it might also be noteworthy that this is not the only time the `requires_grad` flag is touched. Currently there might be a case where a forward hook is registered on the input embeddings which always forces `requires_grad` to true on these.

**Comment by BenjaminBossan** [comment]
Yes, I agree. @NilBiescas would you like to create a PR to update the docstring accordingly?

**Comment by NilBiescas** [comment]
@BenjaminBossan Okey

**Comment by BenjaminBossan** [comment]
Resolved via #2305, thanks.

**Comment by NilBiescas** [comment]
Glad I made my first PR to HF !
